### PR TITLE
FT-10 Add user sign up

### PR DESCRIPTION
**Pull request by boavenn**

### CODE REVIEW

File: src/contexts/auth-context/utils/signUpValidation.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
+import moment from 'moment';
+
+export const validateSignUp = (password, name, birthDate) => {
+    validateName(name);
+    validatePassword(password);
+    validateBirthDate(birthDate);
+}
```
Comment (uzarkov) [inline src/contexts/auth-context/utils/signUpValidation.js:7]: How about validating email as well?

File: src/contexts/auth-context/utils/signUpValidation.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    }
+
+    if (password.search(/[a-z]+/) == -1 || password.search(/[A-Z]+/) == -1 || password.search(/[0-9]/) == -1) {
+        throw new Error("Password has to contain at least one uppercase letter, one lowercase letter and one digit")
+    }
+
+    if (password.search(/[~`!@#$%^&*()_\-+={[}\]\|\:;"'<,>\.?/]+/) == -1) {
+        throw new Error("Password has to contain at least one special character (~`! @#$%^&*()_-+={[}]|\\:;\"'<,>.?/)")
+    }
+}
```
Comment (uzarkov) [inline src/contexts/auth-context/utils/signUpValidation.js:27]: Apart from that, we should check if both password inputs are equal

File: src/contexts/auth-context/utils/signUpValidation.js (hunk 1 above)
Comment (boavenn) [reply src/contexts/auth-context/utils/signUpValidation.js:7]: Regex for email validation looks horrible so I decided to leave its validation to Firebase (if given input is not an email it throws `auth/invalid-email` error or something similar) and I think it should be sufficient

Comment (uzarkov) [reply src/contexts/auth-context/utils/signUpValidation.js:7]: Sure

**Pull request merged**